src/items.rs

```diff
@@ -2886,7 +2886,7 @@ fn rewrite_bounds_on_where_clause(
     let fmt = ListFormatting::new(shape, context.config)
         .tactic(shape_tactic)
         .trailing_separator(comma_tactic)
-        .preserve_newline(true);
+        .preserve_newline(context.config.keep_newline_in_where_clause());
```

Comment:
@calebcartwright originally I thought it would be best to let users configure this, but maybe we'd rather just version gate the fix?